sdk/python/kfp/components/graph_component.py

```diff
@@ -85,3 +89,165 @@ def pipeline_spec(self) -> pipeline_spec_pb2.PipelineSpec:
 
     def execute(self, **kwargs):
         raise RuntimeError('Graph component has no local execution mode.')
+
+    def _dedupe_pipeline_spec(
+        self, pipeline_spec: pipeline_spec_pb2.PipelineSpec
+    ) -> pipeline_spec_pb2.PipelineSpec:
+        """removes duplicated component spec and executor specs caused by tasks
+        calling on the same components.
+
+        component specs are only deduped when the executor specs are the
+        same, this means factors like two tasks called on the same
+        component with differing resource specs would not be deduped,
+        deduping would still occur if the caller tasks differ by their
+        input specifications since this not affect the executor spec of
+        the components
+        """
+        from dataclasses import dataclass
```

Comment:
Per [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html#22-imports), import packages/modules, not classes/functions directly.